### DISCUSSION TITLE
[C# Markup + MVU]: ItemsRepeater with StackLayout does not show items under net8.0-windows10.0.19041

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno.extensions#2291] [Q&A] [answered]
Hi,

I have created a simple app (C# Markup + MVU) that uses `ItemsRepeater` with `StackLayout`.

```cs
    public MainPage() => this
        .DataContext(new BindableMainPageModel(), (page, vm) => page
        .Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))
        .Content(
            new Frame()
            .Background(Colors.LightYellow)
            .Content(MyRepeater(vm))
            )
        );

    private static ItemsRepeater MyRepeater(BindableMainPageModel vm) =>
            new ItemsRepeater()
            .ItemsSource(() => vm.Items)
            .Layout(new StackLayout().Orientation(Orientation.Horizontal).Spacing(10))
            .ItemTemplate<Item>(item
                => new Rectangle()
                .Width(50)
                .Height(30)
                .Stroke(Colors.Black)
                .StrokeThickness(1)
                .RadiusX(4)
                .RadiusY(4)
                .Fill(x => x.Binding(() => item.Color))
            );
```

```cs
internal partial record MainPageModel
{
    public IListState<Item> Items => ListState<Item>.Value(this, () =>
    [
        new Item(){ Color = new SolidColorBrush(Colors.Wheat) },
        new Item(){ Color = new SolidColorBrush(Colors.Orange)  },
        new Item(){ Color = new SolidColorBrush(Colors.Magenta) },
    ]);
}

public class Item
{
    public Brush Color { get; set; } = new SolidColorBrush(Colors.Black);
}
```

This app works as expected when ran on `net8.0-desktop` but fails to draw elements on `net8.0-windows10.0.19041`.
I was thinking the issue is in winui3, that's why I created a winui3 app (XAML) that does the same; and it works as expected.

net8.0-desktop:
![image](https://github.com/unoplatform/uno/assets/54866088/7d673749-4a60-4fc2-b4e7-7b1f8626d5ff)

net8.0-windows10.0.19041
![image](https://github.com/unoplatform/uno/assets/54866088/fd0832b5-0c57-4a83-9791-e0a2f86fd76f)

winui3
![image](https://github.com/unoplatform/uno/assets/54866088/c1c24f97-e8fc-4439-8c58-ab83cf61dc90)

As you can see, on `net8.0-windows10.0.19041` the items do not show up.

Source code for both uno and winui3 apps: [UnoRepeaterApp.zip](https://github.com/unoplatform/uno.extensions/files/15296744/UnoRepeaterApp.zip)

**Comment by agneszitte** [comment]
fyi @dansiegel, @jeromelaban, @kazo0

**Reply by kazo0** [reply]
@dansiegel I think this is the same idea as the `ItemsPanel` for `ItemsControl`/`ListView`/`GridView` and will need a similar change.

**Comment by dansiegel** [comment, marked as answer]
@kucint I'm closing this as I believe this should be resolved by #2404/#2407. Please let me know if you're still having this issue and we can look into more.

**Reply by kucint** [reply]
ok. I will check it and let you know. thanks!
